archive/scripts/analysis_0_single_run/nbconverted/3_visualize_data.py

```diff
@@ -0,0 +1,517 @@
+
+# coding: utf-8
+
+# # Visualize data
+# 
+# In order to start making interpretations we will generate two visualizations of our data
+# 
+# 1. We will verify that the simulated dataset is a good representation of our original input dataset by visually comparing the structures in the two datasets projected onto UMAP space.
+# 
+# 2. We will plot the PCA projected data after adding experiments to examine how the technical variation shifted the data.
+# 
+# 3. We plot the PCA projected data after correcting for the technical variation introduced by the experiments and examine the effectiveness of the correction method by comparing the data before and after the correction.
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+import os
+import ast
+import pandas as pd
+import numpy as np
+import random
+import glob
+from plotnine import (ggplot, 
+                      geom_point,
+                      labs,
+                      aes, 
+                      facet_wrap, 
+                      scale_colour_manual,
+                      guides, 
+                      guide_legend, 
+                      theme_bw, 
+                      theme,  
+                      element_text,
+                      element_rect,
+                      element_line,
+                      element_blank,
+                      ggsave)
+
+from sklearn.decomposition import PCA
+from keras.models import load_model
+import umap
+
+import warnings
+warnings.filterwarnings(action='ignore')
+
+from numpy.random import seed
+randomState = 123
+seed(randomState)
+
+
+# In[2]:
+
+
+# User parameters
+NN_architecture = 'NN_2500_30'
+analysis_name = 'analysis_0'
+num_simulated_samples = 6000
+lst_num_experiments = [1,2,5,10,20,50,100,500,1000,2000,3000,6000]
+
+
+# In[3]:
+
+
+# Load data
+base_dir = os.path.abspath(os.path.join(os.getcwd(),"../.."))    # base dir on repo
+local_dir = "/home/alexandra/Documents"                          # base dir on local machine for data storage
```

Comment:
Perhaps we can talk about this when you are back in the office.  My logic to to store the intermediate files locally was 1) to save storage space in the repo and 2) because the scripts only need the input file to generate everything else.

Reply:
Yes, I do recall telling you 40GB was too much for a github repo. The test data for validating notebooks and results do not have to be the most upstream data in its entirety. It is possible to use some archive services such as figshare or zenodo as needed although they also have storage limitations. What is the minimally sufficient data for you to show the most important results of your project? Is there an intermediate data that can be used to reproduce essential results while requiring <5GB storage?